packages/payment-processor/src/payment/erc20-fee-proxy.ts

```diff
@@ -36,6 +47,34 @@ export async function payErc20FeeProxyRequest(
   return signer.sendTransaction({ data, to, value, ...overrides });
 }
 
+/**
+ * Processes a transaction to pay an ERC20 Request privately with fees.
+ * @param signerOrProvider the Web3 provider, or signer. Defaults to window.ethereum.
+ * @param amount optionally, the amount to pay. Defaults to remaining amount of the request.
+ * @param feeAmount optionally, the fee amount to pay. Defaults to the fee amount.
+ */
+export async function payPrivateErc20FeeProxyRequest(
```

Comment:
Would it be possible to rename this function to include "Hinkal" in the name and move it to a separate file? 

I want to leave room for other private payment types in the future without requiring breaking changes. 

And I want to lay the groundwork for https://github.com/RequestNetwork/requestNetwork/issues/1210

Unnecessary dependency coupling is a long-standing issue with the `payment-processor` package.

```suggestion
export async function payPrivateErc20FeeProxyRequestFromHinkal(
```

Consider renaming the `prepare...()` function as well for consistency.

Reply:
We renamed and moved it to separate folder